pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -682,7 +685,9 @@ private void logBrokerResponse(long requestId, String query, RequestContext requ
           brokerResponse.getOfflineThreadCpuTimeNs(), brokerResponse.getOfflineSystemActivitiesCpuTimeNs(),
           brokerResponse.getOfflineResponseSerializationCpuTimeNs(), brokerResponse.getRealtimeTotalCpuTimeNs(),
           brokerResponse.getRealtimeThreadCpuTimeNs(), brokerResponse.getRealtimeSystemActivitiesCpuTimeNs(),
-          brokerResponse.getRealtimeResponseSerializationCpuTimeNs(), StringUtils.substring(query, 0, _queryLogLength));
+          brokerResponse.getRealtimeResponseSerializationCpuTimeNs(),
+          ((HttpRequesterIdentity) requesterIdentity).getHttpHeaders().toString(),
```

Comment:
Should we just use the host (key,value) from the headers instead of dumping all the headers here? I'm asking because certain header fields (like "accept-language") might not be really useful (at least right now).

Reply:
I agree.  Hence the comment here: https://github.com/apache/pinot/issues/8869#issuecomment-1157729246

Wanted to see what the community feels about this. If we have specific keys then it may only solve for certain set of use cases

Reply:
The idea of keeping the format as "key=value" and then a space and then the query is so that we can write scripts for post-processing, while looking at logs in production easily. If the value has spaces in it, that becomes a problem. It will certainly break some of the scripts we use for debugging.

If we are adding anything like a generic string that may have spaces, I suggest putting a marker AFTER the query, and adding the information there. 

Also, we don't want to make the line too long.

Suggestions to add IP addresses (known not to have spaces in them) is good, and if that suffices then we can always add it in the "key=value" section. But please do name it correctly, instead of calling it "RequesterIdentity". This class can mean different in different environments, so please don't add this in the log message

Reply:
Makes sense. I will change it to have only "X-Forwarded-For" as the key. 

Any recommendations on which delimiter to you use to extract values part of the header. Example : ((HttpRequesterIdentity) requesterIdentity).getHttpHeaders().get("X-Forwarded-For") returns Collection<String> and adding the list to the log might conflict with the `,` separated format